lib/cache.gd

```diff
@@ -82,11 +113,52 @@
 ##  </ManSection>
 ##  <#/GAPDoc>
 ##
-
 DeclareGlobalFunction("MemoizePosIntFunction");
 
+
+#############################################################################
+##
+#F  NEW_SORTED_CACHE(<flushable>)
+##
+##  Set up a cache object suitable for use with `GET_FROM_SORTED_CACHE`.
+##  If `flushable` is `true` (which is also the default), then the cache will
+##  be flushed whenever `FlushCaches` is called.
+##
+BIND_GLOBAL("NEW_SORTED_CACHE",
+function(flushable)
+    local cache;
+    cache := [ [], [] ];
+    if IsHPCGAP then
+        ShareSpecialObj(cache);
+    fi;
+    if flushable then
+        InstallMethod( FlushCaches, [],
+          function()
+              atomic readwrite cache do
```

Comment:
What does `atomic readwrite` do?

Reply:
This is from HPC-GAP, it does nothing in plain GAP, but it takes an exclusive lock on the object `cache` in HPC-GAP

Reply:
Ok thanks for the explanation. So in normal GAP the line `atomic readwrite cache do` and the corresponding `od;` are simply ignored. There is no hidden loop or anything in there?